mirrord-layer/src/lib.rs

```diff
@@ -97,110 +94,129 @@ fn init() {
+                        listen_message.real_port,
+                        ListenData {
+                            port: listen_message.fake_port,
+                            ipv6: listen_message.ipv6,
+                            fd: listen_message.fd,
+                        },
+                    )
+                });
+        }
+    }
+}
+
+async fn handle_daemon_message(
+    daemon_message: DaemonMessage,
+    port_mapping: &mut HashMap<Port, ListenData>,
+    active_connections: &mut HashMap<u16, Sender<TcpTunnelMessages>>,
+) {
+    match daemon_message {
+        DaemonMessage::NewTCPConnection(conn) => {
+            debug!("DaemonMessage::NewTCPConnection {conn:#?}");
+
+            let _ = port_mapping
+                .get(&conn.destination_port)
+                .map(|listen_data| {
+                    let addr = match listen_data.ipv6 {
+                        false => SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), listen_data.port),
+                        true => SocketAddr::new(IpAddr::V6(Ipv6Addr::LOCALHOST), listen_data.port),
+                    };
+
+                    let info =
+                        SocketInformation::new(SocketAddr::new(conn.address, conn.source_port));
+                    {
+                        CONNECTION_QUEUE.lock().unwrap().add(&listen_data.fd, info);
+                    }
+
+                    TcpStream::connect(addr)
+                })
+                .map(|stream| {
+                    let (sender, receiver) = channel::<TcpTunnelMessages>(1000);
+
+                    active_connections.insert(conn.connection_id, sender);
+
+                    task::spawn(async move { tcp_tunnel(stream.await.unwrap(), receiver).await })
+                });
+        }
+        DaemonMessage::TCPData(msg) => {
+            if let Err(fail) = active_connections
+                .get(&msg.connection_id)
+                .map(|sender| sender.send(TcpTunnelMessages::Data(msg.data)))
+                .unwrap()
+                .await
+            {
+                error!("DaemonMessage::TCPData error {fail:#?}");
+                active_connections.remove(&msg.connection_id);
+            }
+        }
+        DaemonMessage::TCPClose(msg) => {
+            if let Err(fail) = active_connections
+                .get(&msg.connection_id)
+                .map(|sender| sender.send(TcpTunnelMessages::Close))
+                .unwrap()
+                .await
+            {
+                error!("DaemonMessage::TCPClose error {fail:#?}");
+                active_connections.remove(&msg.connection_id);
+            }
+        }
+        DaemonMessage::Close => todo!(),
+        DaemonMessage::LogMessage(_) => todo!(),
+    }
+}
+
 async fn poll_agent(mut pf: Portforwarder, mut receiver: Receiver<HookMessage>) {
     let port = pf.take_stream(61337).unwrap(); // TODO: Make port configurable
+
+    // NOTE(alex): `codec` is used to retrieve messages from the daemon (messages that are sent
```

Comment:
docstrings are welcome, but no need to put names ;P

Reply:
Oops, sorry I use an extension to show docs in a tree like view, so tagging them like so helps me with finding stuff.

Do I drop only the `(alex)` part or do you want me to drop the tag (`NOTE`) as well?

Reply:
Yeah I assumed that's some sort of an extension. 
I think it'd be better to drop the tag as well, as most developers probably don't share the same extension and it's pretty implicit that a comment is a note ;p

Reply:
Alright, will drop those.